docs/template_plugin/src/template_infer_request.cpp

```diff
@@ -65,35 +66,51 @@ void TemplateInferRequest::allocateDeviceBuffers() {
     _outputTensors.resize(_networkOutputs.size());
 }
 
+template <typename BlobData, typename GetNetworkPrecisionF>
+static void AllocateImplSingle(BlobMap& blobMap,
+                               BlobMap& networkBlobMap,
+                               BlobData& blobData,
+                               GetNetworkPrecisionF&& GetNetworkPrecision,
+                               const SizeVector& dims) {
+    auto& precision = blobData.second->getTensorDesc().getPrecision();
+    auto layout = blobData.second->getTensorDesc().getLayout();
+    if (dims.size() > 0 && layout == InferenceEngine::Layout::SCALAR) {
+        layout = InferenceEngine::Layout::ANY;
+    }
+    const auto deviceLayout = TensorDesc::getLayoutByDims(dims);
+    Blob::Ptr blob;
+    blob = make_blob_with_precision({precision, dims, layout});
+    blob->allocate();
+    blobMap[blobData.first] = blob;
+
+    auto networkPresion = InferenceEngine::details::convertPrecision(GetNetworkPrecision(blobData.first));
+    Blob::Ptr networkBlob;
+    if (precision == networkPresion && layout == deviceLayout) {
+        networkBlob = blob;
+    } else {
+        networkBlob = make_blob_with_precision({networkPresion, dims, deviceLayout});
+        networkBlob->allocate();
+    }
+    networkBlobMap[blobData.first] = networkBlob;
+}
+
 template <typename BlobDataMap, typename GetNetworkPrecisionF>
 static void AllocateImpl(const BlobDataMap& userDataMap,
                          BlobMap& userBlobMap,
                          BlobMap& deviceBlobMap,
                          GetNetworkPrecisionF&& GetNetworkPrecision,
                          bool isInputBlob = true) {
     for (auto&& userData : userDataMap) {
```

Comment:
If we change `AllocateImplSingle` arguments it means that `userData` can be const